Library/Formula/poco.rb

```diff
@@ -18,18 +19,83 @@ class Poco < Formula
   end
 
   option :cxx11
+  option :universal
 
   depends_on "openssl"
 
   def install
+
     ENV.cxx11 if build.cxx11?
 
-    arch = Hardware.is_64_bit? ? 'Darwin64': 'Darwin32'
-    system "./configure", "--prefix=#{prefix}",
-                          "--config=#{arch}",
-                          "--omit=Data/MySQL,Data/ODBC",
-                          "--no-samples",
-                          "--no-tests"
-    system "make", "install", "CC=#{ENV.cc}", "CXX=#{ENV.cxx}"
+    # the poco project handles most of the architecture specific configuration
+    if build.universal?
+      archs = {
+        "i386" => "Darwin32",
+        "x86_64" => "Darwin64"
+      }
+    elsif MacOS.prefer_64_bit?
+      archs = {
+        "x86_64" => "Darwin64"
+      }
+    else
+      archs = {
+        "i386" => "Darwin32"
+      }
+    end
+
+    arch_dir = ""
+    arch_dirs = []
+
+    archs.each do |arch, arch_mac|
+      if build.universal?
+        # create an architecture specific temporary directory
+        arch_dir = File.join(Dir.pwd, "build-#{arch}")
+        rm_rf arch_dir
+        arch_dirs << arch_dir
+        mkdir arch_dir
+
+        # clean the project before compiling
+        system "make", "clean"
+      end
+
+      # homebrew sets -march=native by default, need to override to compile for i386
+      if arch == "i386"
+        ENV['HOMEBREW_OPTFLAGS'] = '-m32'
+      else
+        ENV['HOMEBREW_OPTFLAGS'] = '-m64'
+      end
```

Comment:
`HOMEBREW_OPTFLAGS` is an implementation detail, not something that should be manipulated by hand.

You probably want to call `ENV.universal_binary` outside of the loop instead.